### PR TITLE
box/iproto: introduce reqstart to iproto_msg

### PR DESCRIPTION
**Pull request by Korablev77**
Subj. Follow-up to 247515e98bf5739d1f4f3b7ace7bc41bc5f191f0

### CODE REVIEW

File: src/box/iproto.cc [hunk 1]
```diff
@@ -356,6 +356,15 @@ struct iproto_msg
...
 	size_t len;
+	/**
+	 * Pointer to the start of unparsed request stored in @a p_ibuf.
+	 * It is used to dump request to flight recorder (if available) in
+	 * TX thread. It is guaranteed that @a reqstart points to the valid
+	 * position: rpos of input buffer is moved after processing the message;
+	 * meanwhile requests are handled in the order they are stored in
+	 * the buffer.
+	 */
+	char *reqstart;
```
Comment (locker) [inline src/box/iproto.cc:367]: Nit: Please make it const. The first argument of `flightrec_write_request` should be const as well.

**Review by locker — CHANGES_REQUESTED** [review]

**Review by alyapunov — APPROVED** [review]

File: src/box/iproto.cc (hunk 1 above)
Comment (Korablev77) [reply src/box/iproto.cc]: I've also added commit which adds extra box include dirs to build main.cc (according to the review of flight recorder).

File: test/app-luatest/prbuf_test.lua [hunk 2]
```diff
@@ -45,3 +46,18 @@ g.test_object_misc = function()
...
 end
+
+g.test_wrong_arg_type = function()
+    local int = ffi.new('int')
+    local map = {}
+    local status, err = pcall(buffer.prbuf_create, int, 4)
+    local expected_err = 'Attempt to prbuf_create%(%) with argument '..
+                         'of wrong type, expected %<char %*%>'
+    t.assert_equals(status, false)
+    t.assert_equals(string.find(tostring(err), expected_err) ~= nil, true)
```
Comment (locker) [inline test/app-luatest/prbuf_test.lua:57]: For future: there's `assert_str_contains` luatest helper.

**Review by locker — APPROVED** [review]

**Comment by locker** [comment]
Please in future, do fix all checkpatch issues or file a ticket in tarantool/checkpatch if there's a false-postivie.

**Pull request merged**